parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -402,6 +402,66 @@ write_files:
     iptables -I FORWARD -d 168.63.129.16 -p tcp --dport 80 -j DROP
     iptables -I FORWARD -d 168.63.129.16 -p tcp --dport 32526 -j DROP
 
+- path: /opt/azure/containers/ensure_imds_restriction.sh
+  permissions: "0755"
+  owner: root
+  content: |
```

Comment:
we don't need to duplicate the script here:

1. add new const to: https://github.com/Azure/AgentBaker/blob/0515c7aee97fc172541691a715ce779ca87cb8e5/pkg/agent/const.go#L43
2. use const and add the b64-encoded script to `cloudInitData` used by the go template: https://github.com/Azure/AgentBaker/blob/0515c7aee97fc172541691a715ce779ca87cb8e5/pkg/agent/variables.go#L20

for example: https://github.com/Azure/AgentBaker/blob/0515c7aee97fc172541691a715ce779ca87cb8e5/parts/linux/cloud-init/nodecustomdata.yml#L262

Reply:
thanks for the suggestion! updated.